src/web/app/src/components/SearchInput/AuthorInput.tsx

```diff
@@ -0,0 +1,116 @@
+import { TextField } from '@material-ui/core';
+import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
+import { Autocomplete } from '@mui/material';
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import { searchServiceUrl } from '../../config';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    input: {
+      fontSize: '14px',
+      color: theme.palette.text.primary,
+      borderRadius: `4rem 4rem 4rem 4rem`,
+      borderColor: theme.palette.info.main,
+      borderWidth: `2px`,
+      transition: theme.transitions.create(['background-color', 'border-color'], {
+        duration: '.5s',
+      }),
+    },
+    listbox: {
+      fontSize: '14px',
+      color: theme.palette.text.primary,
+      backgroundColor: theme.palette.background.paper,
+      '& :hover': {
+        color: theme.palette.text.secondary,
+        backgroundColor: theme.palette.background.default,
+        border: '2px solid',
+      },
+    },
+  })
+);
+
+const fetchResults = async (url: string | null) => {
+  if (url) {
+    const res = await fetch(url);
+    const results = await res?.json();
+    const totalAuthors = results.results;
+    // return up to first 10 results
+    return totalAuthors > 10 ? results.res.slice(0, 10) : results.res;
+  }
+  return [];
+};
+
+interface AuthorInputInterface {
+  text: string;
+  setText: Dispatch<SetStateAction<string>>;
+  labelFor: string;
+}
+
+const AuthorInput = ({ text, setText, labelFor }: AuthorInputInterface) => {
+  const cs = useStyles();
+  const [options, setOptions] = useState<any[]>([]);
```

Comment:
`any` is only used to opt out of typescript typings, let's be explicit for the type here.

Reply:
I have no idea what that means

Reply:
See https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#any

Reply:
I still don't know how you want this changed to

Reply:
Roxanne we should add the type for the typescript file.

Reply:
`useState<any[]>([]);` We can take out `any[]` and put the type instead... Im a little confused on what the type is but I can help you.

Reply:
Yes please, I am a little lost